bmc-common/src/main/java/com/oracle/bmc/ConfigFileReader.java

```diff
@@ -182,18 +194,59 @@ private void accept(String line) {
                                 + trimmedLine);
             }
 
-            final String[] keyValue = trimmedLine.split("=");
-            if (keyValue.length != 2) {
+            // first char was whitespace, this is a multi-line append
+            if (line.charAt(0) != trimmedLine.charAt(0)) {
+                if (lastKey == null) {
+                    throw new IllegalStateException(
+                            "Attempted to define multi-line value without first specifying key: "
+                                    + trimmedLine);
+                }
+                runningMultiLine += "\n" + trimmedLine;
                 return;
             }
 
-            final String key = keyValue[0];
-            final String value = keyValue[1];
+            //
+            final int splitIndex = trimmedLine.indexOf('=');
+            if (splitIndex == -1) {
+                throw new IllegalStateException(
+                        "Found line with no key-value pair: " + trimmedLine);
+            }
+
+            final String key = trimmedLine.substring(0, splitIndex).trim();
+            if (key.isEmpty()) {
+                throw new IllegalStateException("Found line with no key: " + trimmedLine);
+            }
+
+            // flush multi-line before moving to new key
+            flushMultiLine(key);
+
+            final String value = trimmedLine.substring(splitIndex + 1, trimmedLine.length()).trim();
 
             if (!configurationsByProfile.containsKey(currentProfile)) {
                 configurationsByProfile.put(currentProfile, new HashMap<String, String>());
             }
             configurationsByProfile.get(currentProfile).put(key, value);
         }
+
+        private void flushMultiLine(String newLastKey) {
+            if (!runningMultiLine.trim().isEmpty()) {
+                String newValue =
+                        configurationsByProfile.get(currentProfile).get(lastKey) + runningMultiLine;
+                newValue = newValue.trim();
+                configurationsByProfile.get(currentProfile).put(lastKey, newValue);
+            }
+            lastKey = newLastKey;
+            runningMultiLine = "";
+        }
+
+        // call once the entire file has been read
+        private void eof() {
```

Comment:
This same code is repeated in flushMultiLine - might as well call flushMultiLine from eof, or get rid of eof altogether.